erpnext/stock/stock_ledger.py

```diff
@@ -1323,12 +1325,8 @@ def update_qty_in_future_sle(args, allow_negative_stock=False):
 			and warehouse = %(warehouse)s
 			and voucher_no != %(voucher_no)s
 			and is_cancelled = 0
-			and (timestamp(posting_date, posting_time) > timestamp(%(posting_date)s, %(posting_time)s)
-				or (
-					timestamp(posting_date, posting_time) = timestamp(%(posting_date)s, %(posting_time)s)
-					and creation > %(creation)s
-				)
-			)
+			and timestamp(posting_date, time_format(posting_time, %(time_format)s))
+				> timestamp(%(posting_date)s, time_format(%(posting_time)s, %(time_format)s))
```

Comment:
Not sure 100% about this yet, rationale is:
1. update_entries_after updates ALL SLEs that match current timestamp in stripped timestamp format. 
2. So future qty updates should ONLY update entries that are **strictly** greater than posting time.

Reply:
@ankush In this case with the example in the test case, since both transactions have same time stamp, how do we determine the chronology ?

Reply:
Chronology is still determined by these three columns: posting date, posting time, creation


It's highly unlikely and near impossible for two docs to have exact creation time down to milliseconds 😅 (unless it's manually fiddled with)


--

This edge case is happening because user added a new transaction with previous posting time, but frontend doesn't allow exact posting time, so it's before in chronological order even though it's created later. (the same exact test case)



<img width="1281" alt="Screenshot 2022-05-06 at 1 33 25 PM" src="https://user-images.githubusercontent.com/9079960/167092252-112121bd-eb61-4b2f-92ae-97af9b3b5b2b.png">

Reply:
IMO we should strip down posting time to what's visible even when it's not edited manually? This is very confusing.

Reply:
If we were stripping it down to nearest seconds then this confusion and subsequently the bug wouldn't exist 😅

Reply:
removed datetime formatting from this report: 

<img width="1281" alt="Screenshot 2022-05-06 at 1 40 57 PM" src="https://user-images.githubusercontent.com/9079960/167093260-95824aac-78f3-42b1-85c1-fec06b4faa2f.png">